lib/python/script/utils.py

```diff
@@ -467,3 +471,104 @@ def clock():
     if sys.version_info > (3,2):
         return time.perf_counter()
     return time.clock()
+
+
+def legalize_vector_name(name):
+    if not name:
+        raise ValueError("name cannot be empty")
+    if re.match("[^A-Za-z0-9]", name[0]):
+        name[0] = "_"
+    re.sub("[^A-Za-z0-9_]", "_", name)
+    keywords = ["and", "or", "not"]
+    if name in keywords:
+        name = "{name}_".format(**locals())
+    return name
+
+
+def append_node_pid(name):
+    """Add node name and PID to a name (string)
+
+    For the result to be unique, the name needs to be unique within a process.
+    Given that, the result will be unique enough for use in temporary maps
+    and other elements on single machine or an HPC cluster.
+
+    >>> append_node_pid("tmp_raster_1")
+
+    ..note::
+
+        Before you use this function for creating temporary files (i.e., normal
+        files on disk, not maps and other mapset elements), see functions
+        designed for it in the GRASS GIS or standard Python library. These
+        take care of collisions already on different levels.
+    """
+    node = legalize_vector_name(platform.node())
+    pid = os.getpid()
+    return "{name}_{node}_{pid}".format(**locals())
+
+
+def append_uuid(name):
+    """Add UUID4 to a name (string)
+
+    To generate a name of an temporary mapset element which is unique in a
+    system, use :func:`append_node_pid()` in a combination with a name unique
+    within your process.
+
+    To avoid collisions, never shorten the name obtained from this function.
+    A shortened UUID does not have the collision guarantees the full UUID has.
+
+    For a random name of a given shorter size, see :func:`append_random()`.
+
+    >>> append_uuid("tmp")
+
+    ..note::
+
+        See the note about creating temporary files in the
+        :func:`append_node_pid()` description.
+    """
+    suffix = uuid.uuid4().hex
```

Comment:
Nice. I was not aware of the hex attribute in uuid4.

Reply:
I learned about it just recently too, quite handy here, but well, it is in the documentation :-)